### ISSUE TITLE
Custom exception handler

### ISSUE THREAD

**Issue by vrobles-gee** [daniele77/cli#74]
Hi

In recent days, I had and issue when one of my commands did throw an exception without managing it internally. It caused the TelnetServer thread to die, so not only the current session was lost, but we could not connect again until the application was restarted. We already added some fixes at application level, to make sure exceptions are properly catch on every of our commands, as well as some code to warranty a new TelnetServer is started in case the running one dies. 

While doing all this, I was wondering if the library should allow the developer to specify a custom exception handler. The reason for this is not only to avoid relaying on each developer to properly manage the possible exceptions on each command, but also at least three more reasons:

- Avoid duplication of code: right now I am doing the same thing for about 40 different commands, and even if I avoided code duplication by other means, it would be cleaner just have it as part of the library
- Safety: The library could have a default handler that just print to the output stream the exception and can be overriden and/or enabled by the developer. That way even if developer does nothing, exceptions will never make the CLI to crash
- Visibility: On my current solution, I have limited visibility since I am catching the exception inside the command lambda. For instance, I can print the exception on my logs, but I can't print the exact command the user typed to help me reproduce and fix the issue.

So basically I wanted to ask if there is anything already in place that I missed, or any reason to not include the feature. In case you agree with the idea, I am willing to implement that, but would like to ask your opinion about the best place to add that. I was thinking maybe catching the exception inside the Exec function, and just calling a handler registered by the developer on the Cli object if available, or printing an error to the session output stream otherwise.

Thanks

**Comment by daniele77** [comment]
Thank you very much, @vrobles-gee for pointing out that this feature is missing.
While a custom exception handler is a nice feature to have, a library exception handler is instead a must.

I already added both features in the repository. You can find them in the master. There is a unit test and a sample usage in the "complete" example application.

Please, let me know whether it suits your needs and whether the feature can be improved.

Thanks.

**Comment by vrobles-gee** [comment]
Hi @daniele77 I just tested the new feature and it works as expected. I think this issue can be closed now. Thanks for such a quick reply and adding the new code in record time. I really enjoy collaborating in a project managed by someone so open to improvements like you, keep the good work 👍
